### PR TITLE
Array types - hybrid solution

### PR DESCRIPTION
**Pull request by nablaone**
This PR adds another way to handle queries with array columns. If we have an aggregation we can expand (arrayJoin) the array. Computed results looks better. 

Here are examples of how it looks:

Screenshot 2024-09-16 at 17 42 56

Screenshot 2024-09-16 at 17 42 32

### CODE REVIEW

**Comment by jakozaur** [comment]
Great direction, I got some idea how to solve it for good. Though it needs to work with window functions:
https://fiddle.clickhouse.com/082ff22e-b853-45ef-8d96-05f54131ad39

**Comment by jakozaur** [comment]
I debate, but it would probably still make a lot of sense to merge this in its current form.

File: quesma/quesma/schema_array_transformer.go [hunk 1]
```diff
@@ -100,3 +100,87 @@ func NewArrayTypeVisitor(resolver arrayTypeResolver) model.ExprVisitor {
 	}
 	return visitor
 }
+
+func checkIfGoupingByArrayColumn(stm model.SelectCommand, resolver arrayTypeResolver) bool {
```
Comment (trzysiek) [inline quesma/quesma/schema_array_transformer.go:104]: what `stm` stands for?

Comment (trzysiek) [inline quesma/quesma/schema_array_transformer.go:104]:
```suggestion
func checkIfGroupingByArrayColumn(stm model.SelectCommand, resolver arrayTypeResolver) bool {
```

File: quesma/quesma/schema_transformer.go [hunk 2]
```diff
@@ -293,7 +293,13 @@ func (s *SchemaCheckPass) applyArrayTransformations(indexSchema schema.Schema, q
 		return query, nil
 	}
 
-	visitor := NewArrayTypeVisitor(arrayTypeResolver)
+	var visitor model.ExprVisitor
+
+	if checkIfGoupingByArrayColumn(query.SelectCommand, arrayTypeResolver) {
```
Comment (trzysiek) [inline quesma/quesma/schema_transformer.go:298]:
```suggestion
	if checkIfGroupingByArrayColumn(query.SelectCommand, arrayTypeResolver) {
```

File: quesma/quesma/schema_array_transformer.go (hunk 1 above)
Comment (nablaone) [reply quesma/quesma/schema_array_transformer.go:104]: `stm` == `statement`

I'll change to `selectCommand`.

**Review by trzysiek — APPROVED** [review]

**Pull request merged**